flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveBatchScheduler.java

```diff
@@ -937,6 +965,26 @@ SpeculativeExecutionHandler getSpeculativeExecutionHandler() {
         return speculativeExecutionHandler;
     }
 
+    private void tryUpdateResultInfo(IntermediateDataSetID id, DistributionPattern targetPattern) {
```

Comment:
could you add some comments detailing the algorithm and its benefits.

Reply:
+1. e.g. what may have being changed and what should be updated accordingly.

Reply:
updated